working-groups/query-standardization/dsl-survey/kusto-query-language-metrics/README.md

```diff
@@ -0,0 +1,238 @@
+# Kusto Query Language, Metrics Variant DSL Survey Response
+
+### **Short Description:**
+
+The Kusto Query Language for metrics(KQL-M) DSL is a variant of the Kusto Query Language (KQL) DSL that 
+used in Azure Data Explorer, internal to Microsoft focusing on metrics data.
+
+### Overview
+
+1. **List any over-arching design goals for the DSL. For example URI friendliness, minimal syntax,
+   advanced analytical capabilities, etc.**:
+
+   Primary goals were around defining language with user-friendly syntax/semantics and sufficient analytical 
+   capabilities (mostly around data filtering, aggregation, custom computations, and correlations). At 
+   scale performance efficiency was also one of the goals.
+
+1. **What deficiencies or omissions in other DSLs lead to your decision to design a new language?**:
+
+   The primary language of comparison was PromQL, thus one of the goals was to gain corresponding functional 
+   parity. Still, PromQL syntax was considered as a challenge for end users because it required extra learning. 
+   Thus the syntax was inherited from another DBMS language which is very popular internally (it is also exposed 
+   outside as Azure Data Explorer). Also worth mentioning is that PromQL philosophy of 'expression for a single 
+   data point' contradicted to existing data and execution model. Thus the new language expression was 
+   expected to define entire resultant data set (vectors shape).
```

Comment:
Not really clear what 'expression for a single data point' means here. Below you mention that KQL-M is not dealing with tabular data model, so afaiu your data model of a time series is a vector of <ts, number> tuples, with additional map of dimensions attached, same as in Prometheus.

Reply:
I can't answer for Sergey but I thought it was referring to Prom's ability to return the last recorded data point, ala the `query` endpoint.

Reply:
can we create a git issue for this and follow up please?

Reply:
Please correct me if I am wrong, but vector in PromQL semantics will be (unless it is a range vector):

<vm1; 5>
<vm2; 3>
<vm3; 6>
...
<vmN; X>

Our data format upon processing and output is somewhat like below:

<vm1; 5, 3, 1, 9, 2>
<vm2; 3, 2, 0, 3, 1>
<vm3; 6, , , 2, 5>
...
<vmN; X, . . . >